File: Nordeus-Challenge/Assets/_Scripts/Models/Run/ShopOffer.cs
using System;

[Serializable]
public class ShopOffer
{
    public string offerId;
    public string type;    // "ITEM" | "STAT_UPGRADE"
    public int    cost;
    public string itemId;  // ITEM offers only
    public string stat;    // STAT_UPGRADE offers only
    public int    amount;  // STAT_UPGRADE offers only — permanent stat points added
    public bool   purchased;

    public bool IsItem        => type == "ITEM";
    public bool IsStatUpgrade => type == "STAT_UPGRADE";
}

File: backend/src/main/java/com/example/backend/models/encounters/ShopOffer.java
package com.example.backend.models.encounters;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class ShopOffer {

    private String offerId;
    private OfferType type;
    private int cost;

    /** Item id — set only when type == ITEM. */
    private String itemId;

    /** Stat name ("health","attack","defense","magic") — set only when type == STAT_UPGRADE. */
    private String stat;

    /** Stat point amount — set only when type == STAT_UPGRADE. */
    private int amount;

    private boolean purchased;

    public enum OfferType {
        ITEM,
        STAT_UPGRADE
    }
}
